### PR TITLE
UPDATE: TPT3-80. Filter cases by start date and end date while retrieving them by staff person id. Updated test

### PR DESCRIPTION
**Pull request by denys-davydov**
**Merge after** https://github.com/ca-cwds/api-core/pull/66

### CODE REVIEW

File: src/main/java/gov/ca/cwds/cm/service/dto/ReferralDTO.java [hunk 1]
```diff
@@ -13,9 +13,11 @@
 import javax.validation.constraints.NotNull;
 import javax.validation.constraints.Size;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 
 /** @author CWDS TPT-3 Team */
 @Data
+@EqualsAndHashCode(callSuper = false)
```
Comment (zemlyanoi) [inline src/main/java/gov/ca/cwds/cm/service/dto/ReferralDTO.java]: What the reason to add here EqualsAndHashCode methods ? This Class is not used in Map collections and as I remember it is not necessary equal this type of objects

File: src/main/java/gov/ca/cwds/cm/service/dto/facade/ReferralByStaff.java [hunk 2]
```diff
@@ -6,11 +6,13 @@
 import gov.ca.cwds.data.legacy.cms.entity.enums.AssignmentType;
 import gov.ca.cwds.rest.api.Response;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 
 /**
  * @author CWDS TPT-3 Team
  */
 @Data
+@EqualsAndHashCode(callSuper = true)
```
Comment (zemlyanoi) [inline src/main/java/gov/ca/cwds/cm/service/dto/facade/ReferralByStaff.java]: The same question

File: src/main/java/gov/ca/cwds/cm/service/dto/ReferralDTO.java (hunk 1 above)
Comment (denys-davydov) [reply src/main/java/gov/ca/cwds/cm/service/dto/ReferralDTO.java]: @Data already includes @EqualsAndHashCode. 
But there were warnings during project compilation, that superclass of ReferralDTO is not used in equals() and hashcode(). If it is a mistake you have to override @EqualsAndHashCode annotation and use callSuper = true as a parameter. If it is made intentionally then you should reflect it by explicit overriding of @EqualsAndHashCode(callSuper = false).

File: src/main/java/gov/ca/cwds/cm/service/dto/facade/ReferralByStaff.java (hunk 2 above)
Comment (denys-davydov) [reply src/main/java/gov/ca/cwds/cm/service/dto/facade/ReferralByStaff.java]: The same answer as above

**Pull request merged**